utils.py

```diff
@@ -157,3 +159,22 @@ def balance(it):
 
                 last_label = e[1]
                 yield e
+
+
+def download(url, filename, block='█'):
+    with open(filename, 'wb') as file:
+            response = requests.get(url, stream=True)
+            total = response.headers.get('content-length')
+
+            if total is None:
+                file.write(response.content)
+            else:
+                downloaded = 0
```

Comment:
Nit: No need to define this here

Reply:
I think we do need to define this here to be zero because after that we're going to increment it as

```python
downloaded += len(data)
```

Reply:
I don't know why I had missed the `+` 😄